weather_app.py

```diff
@@ -0,0 +1,50 @@
+import requests
+
+# Data Source Class
+class HourlyTemperature:
+    def __init__(self, temperatures: list):
+        self.temperatures = temperatures
+
+# Service Class
+class WeatherService:
```

Comment:
This class is supposed to be implemented using abstraction and interfaces (i.e. `AbstractWeatherService` defining the methods, and `ApiWeatherService` & `MockedWeatherService` extending this interface with actual implementations).